src/operator/nn/mkldnn/mkldnn_base.cc

```diff
@@ -237,13 +237,14 @@ mkldnn_memory_format_t GetDefaultFormat(mkldnn::memory::desc desc) {
       case mkldnn_gOIhw16o16i:
       case mkldnn_gIOhw16o16i:
       case mkldnn_gOihw8o:
+      case mkldnn_Goihw8g:
       case mkldnn_gOihw16o:
       case mkldnn_gOhwi8o:
       case mkldnn_gOhwi16o:
       case mkldnn_gOhIw16o4i:
         return mkldnn_goihw;
       default:
-        LOG(FATAL) << "Unknown MKLDNN format for 4 dimensions: " << desc.data.format;
+        LOG(FATAL) << "Unknown MKLDNN format for 5 dimensions: " << desc.data.format;
```

Comment:
Is there a test for this

Reply:
We notice that there is a test case failed here and it was caused by a missing data format "mkldnn_Goihw8g". Then we found the error message is misleading.
Please find the error message here: http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/PR-9918/4/pipeline/513

Reply:
Aah great. But the error message is indeed pretty misleading. Would it be possible to make a better fitting test?

Reply:
Do you mean a negative test case which can run into the default statement?

Reply:
Data format is an attribute of mkldnn memory which has been hidden into NDArray. So I don't think the default statement could be touched from a python test case currently. Actually, all mkldnn data formats are covered by the list here now. We provided the default statement just for improving the robustness of the code. But there is still a risk that mkldnn may add other format into its own [data format list](https://github.com/intel/mkl-dnn/blob/master/include/mkldnn_types.h#L107) in the future and that will run into this default statement then.
If we should set a CPP test case for it, could you kindly point out where I can find the examples for this kind of test cases. Thanks.

Reply:
Because all the 5D data formats that mkldnn supported currently are covered by the swich-case statement here. So I don't think we can create a `mkldnn::memory::desc` with another 5D data format to touch the default statement.
One feasible method is to set a test case to check whether there is a new format added into mkldnn data format list. If the test case is failed, maybe we need check the code here and add the new format into the switch-case statement.

Reply:
This sounds like a good idea! Especially considering that updating the MKLDNN submodule requires a PR, problems like this would be caught during an early stage.

Reply:
Did you create the jira issue?

Reply:
Created. [MXNET-98](https://issues.apache.org/jira/browse/MXNET-98)